samplers/jaegerremote/sampler_remote.go

```diff
@@ -0,0 +1,313 @@
+func (s *Sampler) UpdateSampler() {
+	res, err := s.samplingFetcher.Fetch(s.serviceName)
+	if err != nil {
+		//s.logger.Infof("failed to fetch sampling strategy: %v", err)
+		return
+	}
+	strategy, err := s.samplingParser.Parse(res)
+	if err != nil {
+		//s.logger.Infof("failed to parse sampling strategy response: %v", err)
+		return
+	}
+
+	s.Lock()
+	defer s.Unlock()
+
+	if err := s.updateSamplerViaUpdaters(strategy); err != nil {
+		//s.logger.Infof("failed to handle sampling strategy response %+v. Got error: %v", res, err)
+		return
+	}
+}
+
+// NB: this function should only be called while holding a Write lock
+func (s *Sampler) updateSamplerViaUpdaters(strategy interface{}) error {
+	for _, updater := range s.updaters {
+		sampler, err := updater.Update(s.sampler, strategy)
+		if err != nil {
+			return err
+		}
+		if sampler != nil {
+			s.sampler = sampler
+			return nil
+		}
+	}
+	return fmt.Errorf("unsupported sampling strategy %+v", strategy)
+}
+
+// -----------------------
+
+// probabilisticSamplerUpdater is used by Sampler to parse sampling configuration.
+type probabilisticSamplerUpdater struct{}
+
+// Update implements Update of samplerUpdater.
+func (u *probabilisticSamplerUpdater) Update(sampler trace.Sampler, strategy interface{}) (trace.Sampler, error) {
+	type response interface {
+		GetProbabilisticSampling() *jaeger_api_v2.ProbabilisticSamplingStrategy
+	}
+	var _ response = new(jaeger_api_v2.SamplingStrategyResponse) // sanity signature check
+	if resp, ok := strategy.(response); ok {
+		if probabilistic := resp.GetProbabilisticSampling(); probabilistic != nil {
+			if ps, ok := sampler.(*probabilisticSampler); ok {
+				if err := ps.Update(probabilistic.SamplingRate); err != nil {
+					return nil, err
+				}
+				return sampler, nil
+			}
+			return newProbabilisticSampler(probabilistic.SamplingRate), nil
+		}
+	}
+	return nil, nil
+}
+
+// -----------------------
+
+// rateLimitingSamplerUpdater is used by Sampler to parse sampling configuration.
+type rateLimitingSamplerUpdater struct{}
+
+// Update implements Update of samplerUpdater.
+func (u *rateLimitingSamplerUpdater) Update(sampler trace.Sampler, strategy interface{}) (trace.Sampler, error) {
+	type response interface {
+		GetRateLimitingSampling() *jaeger_api_v2.RateLimitingSamplingStrategy
+	}
+	var _ response = new(jaeger_api_v2.SamplingStrategyResponse) // sanity signature check
+	if resp, ok := strategy.(response); ok {
+		if rateLimiting := resp.GetRateLimitingSampling(); rateLimiting != nil {
+			rateLimit := float64(rateLimiting.MaxTracesPerSecond)
+			if rl, ok := sampler.(*rateLimitingSampler); ok {
+				rl.Update(rateLimit)
+				return rl, nil
+			}
+			return newRateLimitingSampler(rateLimit), nil
+		}
+	}
+	return nil, nil
+}
+
+// -----------------------
+
+// adaptiveSamplerUpdater is used by Sampler to parse sampling configuration.
+// Fields have the same meaning as in perOperationSamplerParams.
+type adaptiveSamplerUpdater struct {
```

Comment:
nit
```suggestion
type perOperationSamplerUpdater struct {
```

Reply:
Can you give me some write permission to modify these PR

@yurishkuro @kvrhdn

Reply:
I am not aware of how to do that. If the original author is unavailable to complete this, I would suggest branching off `kvrhdn:jaeger_remote` and creating a new PR (keep the original author's commits).